src/lava/lib/optimization/utils/solver_tuner.py

```diff
@@ -1,76 +1,164 @@
 # Copyright (C) 2022 Intel Corporation
 # SPDX-License-Identifier: BSD-3-Clause
 # See: https://spdx.org/licenses/
+
 import itertools as it
 import typing as ty
 
 import random
+import numpy as np
 
 
 class SolverTuner:
-    """Class to find and set hyperparameters for an OptimizationSolver."""
+    """Utility class to optimize hyper-parameters by random search."""
 
-    def __init__(self, params_grid: ty.Dict):
+    def __init__(self,
+                 search_space: list,
+                 params_names: list,
+                 shuffle: bool = False,
+                 seed: int = 0):
         """
-        Instantiate a SolverTuner, defining the admissible search space.
+        Instantiate a SolverTuner, defining a search space and a list of
+        hyperparameters names.
 
         Parameters
         ----------
-        params_grid: dict
-            Dictionary where the keys are the name of the hyperparameters to be
-            tuned, and the values are tuples containing all the admissible
-            values for the associated hyperparameter.
+        search_space
+            List of hyperparameter tuple to evaluate.
+        params_names
+            List of hyperparameters names, one for each tuple dimension.
+        shuffle
+            Boolean flag to control search space shuffling. If set to False,
+            the order of search_space list is preserved, and can be useful to
+            prioritize the evaluation of certain hyperparamters tuples.
+        seed
+            Seed for random shuffling and numpy seeding.
         """
-        self._params_grid = params_grid
+        self._search_space = list(search_space)
+        self._params_names = params_names
+        self._shuffle = shuffle
+        self._seed = seed
+
+        if self._shuffle:

```

Comment:
Minor code style: I would add the definition of the search space and it's shuffling to the tune method. No need to re-instantiate the SolverTuner each time we want a new set of Hyperparameters or change the seed.